src/quote/Quote.ts

```diff
@@ -0,0 +1,354 @@
+import { AxiosError, AxiosRequestConfig } from 'axios';
+
+import productTypes from '../../generated/product-types.json';
+import products from '../../generated/products.json';
+import { ProductTypes } from '../../generated/types';
+import {
+  COMMISSION_DENOMINATOR,
+  CoverAsset,
+  DEFAULT_COMMISSION_RATIO,
+  DEFAULT_SLIPPAGE,
+  MAXIMUM_COVER_PERIOD,
+  MINIMUM_COVER_PERIOD,
+  NEXUS_MUTUAL_DAO_TREASURY_ADDRESS,
+  PaymentAsset,
+  SLIPPAGE_DENOMINATOR,
+  TARGET_PRICE_DENOMINATOR,
+} from '../constants';
+import { Ipfs } from '../ipfs';
+import { NexusSDKBase } from '../nexus-sdk-base';
+import {
+  CoverRouterProductCapacityResponse,
+  CoverRouterQuoteResponse,
+  ErrorApiResponse,
+  GetQuoteApiResponse,
+  GetQuoteResponse,
+  GetQuoteAndBuyCoverInputsParams,
+  NexusSDKConfig,
+  QuoteParams,
+  IPFSTypeContentTuple,
+  IPFS_CONTENT_TYPE_BY_PRODUCT_TYPE,
+} from '../types';
+
+type ProductDTO = Omit<(typeof products)[number], 'productType'> & {
+  productType: ProductTypes;
+};
+
+const productsMap: Record<number, ProductDTO> = products.reduce(
+  (acc, product) => ({ ...acc, [product.id]: product }),
+  {},
+);
+
+/**
+ * Class for handling quote-related functionality
+ */
+export class Quote extends NexusSDKBase {
+  private ipfs: Ipfs;
+
+  /**
+   * Create a new Quote instance
+   * @param config SDK configuration
+   * @param ipfs IPFS instance for content upload and validation
+   */
+  constructor(config: NexusSDKConfig = {}, ipfs?: Ipfs) {
+    super(config);
+    this.ipfs = ipfs || new Ipfs(config);
+  }
+
+  /**
+   * Get quote and buy cover inputs
+   * @param params Parameters for the quote
+   * @returns Quote and buy cover inputs
+   */
+  public async getQuoteAndBuyCoverInputs(
+    params: GetQuoteAndBuyCoverInputsParams,
+  ): Promise<GetQuoteApiResponse | ErrorApiResponse> {
+    const {
+      productId,
+      amount,
+      period,
+      coverAsset,
+      buyerAddress,
+      slippage = DEFAULT_SLIPPAGE / SLIPPAGE_DENOMINATOR,
+      ipfsCidOrContent = '',
+      paymentAsset = coverAsset,
+      coverId = 0,
+    } = params;
```

Comment:
There were 2 more optional params for commission details: `commissionRatio` and `commissionDestination`

```
 * @param {number} commissionRatio - The commission ratio for the cover purchase (optional).
 * @param {Address} commissionDestination - The address to which the commission is sent (optional).
 ```